client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -17,30 +17,43 @@ import {
 	transformMailboxForCart,
 	validateMailboxes,
 } from 'calypso/lib/titan/new-mailbox';
+import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
+import type { TranslateResult } from 'i18n-calypso';
+import type { ChangeEvent } from 'react';
 
 import './style.scss';
 
-const noopWithCallback = ( ignored, callback = () => {} ) => {
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noopWithCallback = ( cartItem: MinimalRequestCartProduct, callback = () => {} ) => {
 	callback();
 };
 
-const ProfessionalEmailFeature = ( { children } ) => {
+const ProfessionalEmailFeature = ( { children }: { children: TranslateResult } ) => {
 	return (
 		<li>
-			<Gridicon icon="checkmark" size="18" />
+			<Gridicon icon="checkmark" size={ 18 } />
```

Comment:
Just curious: did you have to change that because you've converted this component to TypeScript?

Reply:
Yes -- TS requires a number for the value.